From: Emilio Pozuelo Monfort <pochu@debian.org>
Date: Thu, 12 Sep 2019 09:26:05 +0100
Subject: tests-collection-add-setup-delay

Bug: https://bugzilla.gnome.org/show_bug.cgi?id=779041
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=837067

Update by Carsten Leonhardt <leo@debian.org>: move sleep(1) to the
beginning of teardown() to address bug #855951
---
 libsecret/test-collection.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libsecret/test-collection.c b/libsecret/test-collection.c
index 662c41d..dd50de4 100644
--- a/libsecret/test-collection.c
+++ b/libsecret/test-collection.c
@@ -56,12 +56,15 @@ setup (Test *test,
 	test->service = secret_service_get_sync (SECRET_SERVICE_NONE, NULL, &error);
 	g_assert_no_error (error);
 	g_object_add_weak_pointer (G_OBJECT (test->service), (gpointer *)&test->service);
+
+	sleep(1);
 }
 
 static void
 teardown (Test *test,
           gconstpointer unused)
 {
+	sleep(1);
 	g_object_unref (test->service);
 	secret_service_disconnect ();
 	g_assert_null (test->service);
